src/menu/create_wallet_menu.c

```diff
@@ -0,0 +1,181 @@
+ * limitation fees for hosting or consulting/ support services related
+ * to the Software), a product or service whose value derives, entirely
+ * or substantially, from the functionality of the Software. Any license
+ * notice or attribution required by the License must also include
+ * this Commons Clause License Condition notice.
+ *
+ * Software: All X1Wallet associated files.
+ * License: MIT
+ * Licensor: HODL TECH PTE LTD
+ *
+ ******************************************************************************
+ */
+
+/*****************************************************************************
+ * INCLUDES
+ *****************************************************************************/
+#include "create_wallet_menu.h"
+
+#include "constant_texts.h"
+#include "create_wallet_flow.h"
+#include "menu_priv.h"
+#include "ui_screens.h"
+
+/*****************************************************************************
+ * EXTERN VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * PRIVATE MACROS AND DEFINES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * PRIVATE TYPEDEFS
+ *****************************************************************************/
+typedef enum {
+  GENERATE_NEW_WALLET = 1,
+  RESTORE_FROM_SEED,
+} create_wallet_options_e;
+
+/*****************************************************************************
+ * STATIC FUNCTION PROTOTYPES
+ *****************************************************************************/
+/**
+ * @brief This is the initializer callback for the create wallet menu.
+ *
+ * @param ctx The engine context* from which the flow is invoked
+ * @param data_ptr Currently unused pointer set by the engine
+ */
+static void create_wallet_menu_initialize(engine_ctx_t *ctx,
+                                          const void *data_ptr);
+
+/**
+ * @brief This is the UI event handler for the create wallet menu.
+ * @details The function decodes the UI event and calls the create wallet flow
+ * based on selection. It deletes the create wallet menu step if the back button
+ * is pressed on the menu
+ *
+ * @param ctx The engine context* from which the flow is invoked
+ * @param ui_event The ui event object which triggered the callback
+ * @param data_ptr Currently unused pointer set by the engine
+ */
+static void create_wallet_menu_handler(engine_ctx_t *ctx,
+                                       ui_event_t ui_event,
+                                       const void *data_ptr);
+/*****************************************************************************
+ * STATIC VARIABLES
+ *****************************************************************************/
+const flow_step_t create_wallet_step = {
+    .step_init_cb = create_wallet_menu_initialize,
+    .p0_cb = NULL,
```

Comment:
p0 handler missing

Reply:
what is the intended behavior of P0 here anyway?

Reply:
I think on menu, P0 event is not expected.. so I have increased P0 timeout to `INFINITE_WAIT_TIMEOUT`.

As a precautionary step (in the case P0 ever occurs), I have added the callback to clear p0  event